src/extended_range_proof.rs

```diff
@@ -99,13 +100,19 @@ pub trait ExtendedRangeProofService {
 
 /// Extended blinding factor vector used as part of the witness to construct an extended proof, or rewind data
 /// extracted from a range proof containing the mask (e.g. blinding factor vector).
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Zeroize)]
 pub struct ExtendedMask<K>
 where K: SecretKey
 {
     secrets: Vec<K>,
 }
 
+impl<K:SecretKey> Drop for ExtendedMask<K>{
+    fn drop(&mut self) {
+        self.secrets.zeroize();
+    }
+}
```

Comment:
The `SecretKey` trait already enforces zeroizing on drop, so I don't think this is necessary.

Reply:
If you still wanted to separately enforce zeroizing on drop at the `ExtendedMask` level as a safeguard against future structural changes, you should be able to simply derive `ZeroizeOnDrop` instead of this manual implementation.

Reply:
See [this branch](https://github.com/AaronFeickert/tari-crypto/commit/0a7e629c4cf4fa84af0957febbd9a62c1b4741f5) for the derived traits.

Reply:
Also, if you derive `ZeroizeOnDrop`, I'd also do `Zeroize` too; this will allow a caller to manually call `zeroize` if they really want to. You don't get `Zeroize` for free for `ZeroizeOnDrop`.